packages/react/src/file-upload/FileUpload.tsx

```diff
@@ -102,28 +104,105 @@ export const FileUpload = forwardRef<HTMLInputElement, FileUploadProps>(
 		const maxSizeBytes = maxSize && !isNaN(maxSize) ? maxSize * 1000 : 0;
 		const formattedMaxFileSize = formatFileSize(maxSizeBytes);
 
-		const [fileRejections, setFileRejections] = useState<RejectedFile[]>([]);
+		if (maxFiles !== undefined && maxFiles < 1) {
+			throw new Error('maxFiles cannot be less than 1');
```

Comment:
I do think this should be a `console.warn`. Or at least only throw in non-production envs. What do you think?

Reply:
Also, as a general rule, I feel it's nice to give some context to where the error/warning is originating. What do you think about prefixing the warning with the component name?

Reply:
Changed it to console.warn.('FileUpload: maxFiles cannot be less than 1. The property is being ignored.')

I checked and if `maxFiles={0}` then the code treats maxFiles as if it is undefined.